src/customJs/tools/promo_code/PromoCodeViewer.test.js

```diff
@@ -0,0 +1,27 @@
+import { React } from '../../unlayer-react';
+import { render, screen } from '@testing-library/react';
+import { PromoCodeViewer } from './PromoCodeViewer';
+
+describe(PromoCodeViewer.name, () => {
+  const rest = {
+    isViewer: true,
+    toolInfo: {
+      icon: 'toolIconPath',
+      label: 'toolLabel',
+      name: 'toolName',
+    },
+  };
+  /* TODO: validate logic to show empty view */
+  it('must be render the empty content for unselect promotion code options ', async () => {
```

Comment:
What do you think about adding a new test for when the promo-code is selected?

Reply:
Something like _should not render the empty content when a promotion code is selected_.

Reply:
sure but I think would be better when the view will be done, right now the view is a pre with the JSON

Reply:
Well, you can reduce your test to your code: _when there is a selected promotion code the empty viewer should not be rendered_